Robus/src/msgAlloc.c

```diff
@@ -17,45 +18,134 @@
  ******************************************************************************/
 
 //msg allocation management
-uint8_t msg_buffer[MSG_BUFFER_SIZE];     /*!< Memory space used to save and alloc messages. */
-msg_allocator_t alloc_table[MAX_MSG_NB]; /*!< Message allocation table. */
-uint16_t alloc_id;                       /*!< last writen alloc_table id. */
-msg_t *current_msg;                      /*!< current work in progress msg pointer. */
-uint8_t *data_ptr;                       /*!< Pointer to the next data able to be writen into msgbuffer. */
+volatile uint8_t msg_buffer[MSG_BUFFER_SIZE];      /*!< Memory space used to save and alloc messages. */
+volatile msg_allocator_t alloc_table[MAX_MSG_NB];  /*!< Message allocation table. */
+volatile uint16_t alloc_id;                        /*!< last writen alloc_table id. */
+volatile msg_t *current_msg;                       /*!< current work in progress msg pointer. */
+volatile uint8_t *data_ptr;                        /*!< Pointer to the next data able to be writen into msgbuffer. */
+vm_t *module_concerned_by_current_msg[MAX_MSG_NB]; /*!< Save the module concerned by the current message. */
```

Comment:
We could remove this table and just write modules pointer into alloc table with a 0 on msg pointer. We use this 0 pointer on the message to know if the slot is free or not.
But if we do this it means to potentially remove occupied allocator slots before the end of the Msg. So luos loose the time to receive the end of the data to manage them.

Reply:
The size of this table should not be MAX_MSG_NB but MAX_MODULE_NUMBER.